### ISSUE TITLE
dotNet Maui UI Automation test samples for Android do not work

### ISSUE THREAD

**Issue by SiliconKidZA** [dotnet/maui-samples#500]
In BasicAppiumNUnitSample the Android automation tests do not work with the latest version of Appium and the latest version of the uiautomator2 appium driver.

When attempting to run the Android tests the following error occurs:

```OneTimeSetUp: OpenQA.Selenium.WebDriverException : An unknown server-side error occurred while processing the command. Original error: Cannot start the 'com.companyname.basicappiumsample' application. Consider checking the driver's troubleshooting documentation. Original error: Cannot start the 'com.companyname.basicappiumsample' application. Consider checking the driver's troubleshooting documentation. Original error: Activity name '.com.companyname.basicappiumsample.MainActivity' used to start the app doesn't exist or cannot be launched! Make sure it exists and is a launchable activity```

In C# the main activity is declared as follows:

```
[Activity(Theme = "@style/Maui.SplashTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize | ConfigChanges.Density)]
[Register("com.companyname.basicappiumsample.MainActivity")]
public class MainActivity : MauiAppCompatActivity
{
}
```

And the UiAutomator2 driver is initialised in C# as follows:

```androidOptions.AddAdditionalAppiumOption(AndroidMobileCapabilityType.AppActivity, "com.companyname.basicappiumsample.MainActivity");```

When looking carefully at the error message, It appears as though a full stop is being prefixed to the name of the main activity specified and I am assuming that is the root cause of the problem.

I have downloaded, compiled, and run the samples exactly as they are without any modifications and I have also verified this same failure behaviour in another unrelated Maui project.

My best guess would be that the UiAutomator2 driver changed somewhere along the line and broke this functionality but I'm really not sure.

I have tried earlier versions of the driver but have not found a version that works yet.

**Comment by jfversluis** [comment]
Did you deploy the app before trying to run the tests on them? If the app isn't deployed yet, it's not able to start it by the identifier.

**Comment by SiliconKidZA** [comment]
@jfversluis :  Thank you for pointing me in the right direction, I've solved it.

Our enterprise app was deployed to the emulator already and running BUT the version deployed to the emulator did not have the Register attribute built into it, and I was so deep in the forest that I couldn't see the wood for the trees anymore.

The sample app wasn't working because of exactly what you said, I had never actually deployed it to the emulator and run it standalone prior to attempting to run the tests, so there was no APK on device yet, which in retrospect is obviously a problem and not going to work, but there you have it.

Your comment was exactly what I needed to fill in the missing puzzle piece.

Thank you.
